tests/backend/core/Core/Unit/Logic/Segment/RpcBulkEditor/SegmentBulkEditorServiceTest.php

```diff
@@ -0,0 +1,161 @@
+        $this->entityManager = $this->getContainer()->get(EntityManagerInterface::class);
+        $this->entityType = $this->entityManager->getClassMetadata(Segment::class)->getName();
+        $this->user = $this->loginTestUser();
+        $this->prepareSegments();
+    }
+
+    private function prepareSegments(): void
+    {
+        $em = $this->getEntityManager();
+        $this->segment1->setRecommendation('Initial text 1');
+        $this->segment2->setRecommendation('Initial text 2');
+        $em->persist($this->segment1);
+        $em->persist($this->segment2);
+        $em->flush();
+    }
+
+    public function testUpdateSegmentsWithNotNullAssignee(): void
+    {
+        self::assertNull($this->segment1->getAssignee());
+        self::assertNull($this->segment2->getAssignee());
+
+        $this->sut->updateSegments([$this->segment1, $this->segment2], [], [], $this->user, null);
+
+        self::assertEquals($this->user->getId(), $this->segment1->getAssignee()->getId());
+        self::assertEquals($this->user->getId(), $this->segment2->getAssignee()->getId());
+    }
+
+    public function testUpdateSegmentsWithNullAssignee(): void
+    {
+        $this->segment1->setAssignee($this->user);
+        $this->segment2->setAssignee($this->user);
+
+        self::assertEquals($this->user->getId(), $this->segment1->getAssignee()->getId());
+        self::assertEquals($this->user->getId(), $this->segment2->getAssignee()->getId());
+
+        $this->sut->updateSegments([$this->segment1, $this->segment2], [], [], null, null);
+
+        self::assertNull($this->segment1->getAssignee());
+        self::assertNull($this->segment2->getAssignee());
+    }
+
+    public function testDetectValidAssignee(): void
+    {
+        $user = $this->loginTestUser();
+        $assignee = $this->sut->detectAssignee($user->getId());
+        self::assertEquals($user->getId(), $assignee->getId());
+    }
+
+    public function testDetectInvalidIdAssignee(): void
+    {
+        self::expectException(UserNotFoundException::class);
+        $invalidAssigneeId = '123';
+        $this->sut->detectAssignee($invalidAssigneeId);
+    }
+
+    public function testDetectNullAssignee(): void
+    {
+        $assignee = $this->sut->detectAssignee(null);
+        self::assertNull($assignee);
+    }
+
+    public function testGetValidSegments()
+    {
+        $segments = $this->sut->getValidSegments([$this->segment1->getId(),  $this->segment2->getId()], $this->procedure->getId());
+
+        self::assertContains($this->segment1, $segments);
+        self::assertContains($this->segment2, $segments);
+    }
+
+    public function testGetInvalidSegments()
+    {
+        $invalidProcedureId = '123';
+        self::expectException(InvalidArgumentException::class);
+        $segments = $this->sut->getValidSegments([$this->segment1->getId(), $this->segment2->getId()], $invalidProcedureId);
```

Comment:
self::assertNotContains($this->segment1, $segments);
self::assertNotContains($this->segment2, $segments);

Reply:
done bb2fefc7b61d51f164aeac8f977a779a7f312917